src/components/Registration/index.jsx

```diff
@@ -0,0 +1,97 @@
+import React, {useState, useEffect} from 'react';
+import './style.css';
+import userIcon from './img/user-solid.svg'
+
+const Registation = () => {
+
+    const [user, setUser] = useState({
+        email: "",
+        username: "",
+        password: "",
+        passwordConfirm: "",
+      });
+
+    const handleSubmit= (e) => {
+        e.preventDefault();
+        if (user.password === user.passwordConfirm && user.username !== '' && user.email !== '' && user.password !== '' && user.passwordConfirm !== '') {
+            console.log(user)
+          } else if (user.password !== user.passwordConfirm) {
+            alert('Your password and confirmation password do not match.')
+          } else {
+            alert('Please fill out all required fields.')
+          }
+    };
+
+    useEffect(() => {
+        if (user.email.indexOf("@") >= 0 && user.username.length === 0) {
+          setUser({ ...user, username: user.email.substring(0, user.email.indexOf("@")) });
+        }
+      }, [user.email]);
```

Comment:
U formulářů je vždycky strašný drbání s tím kdy tyhle akce provést :D Popíšu ti proč (z UX hlediska. technicky to funguje):

ty to provádíš na to když uživatel klikne na "Register" button. Ale on neví že to políčko vyplníš za něj, tj před tím než klikne na Registrovat, tak vyplní i username, tj mu tím nepomůžeš. Jako řešení by mohlo být že to k tomu políčku napíšeš aby to uživatel věděl. (Navíc tady to uživateli přepíšeš i pokud tam něco sám napsal, not good.)

To ale není ideální protože pak by každý formulář měl složité instrukce. Tak můžeš udělat to že tenhle check a modifikaci provedeš na onChange email políčka. Tím pádem když uživatel začne vyplňovat email, pokud je správný, tak se mu začne samo vyplňovat i jméno. Pokud ho bude chtít změnit, tak ho pak přepíše a je happy.

Ale to že se mu bude formulář měnit pod rukama mu zase bude krást pozornost jak bude psát, tj ideálně to udělat až když to políčko dopíše nebo když nebude nějakou dobu psát (třeba 500ms).

uff, z fomulářema je vždycky drbání aby byly dobře :D